feat(queues): allow to batch pull requests (#3012)
Mergifyio/mergify-engine: docs/source/actions/queue.rst

       between 1 and 20.
       See :ref:`speculative checks`.

   * - ``batch_size``
     - int
     - 1

>      - |premium plan tag| The maximum number of pull requests per speculative checks in the queue. Must be
>        between 1 and 20.

Reviewer: ```suggestion
     - |premium plan tag| The maximum number of pull requests per speculative check in the queue. Must be
       between 1 and 20.
```

of the CI in parallel.

In the following example, by setting the ``speculative_checks`` option to

> ``2``, Mergify will create up to new 2 pull requests to check if the first

Reviewer: ```suggestion
``2``, Mergify will create up to 2 new pull requests to check if the first
```

three enqueued pull requests are mergeable.

.. code-block:: yaml

    queue_rules:
      - name: default
        speculative_checks: 2
        conditions:
          - check-success=Travis CI - Pull Request

    pull_request_rules:
      - name: merge using the merge queue and speculative checks
        conditions:
          - base=main
          - "#approved-reviews-by>=2"
          - check-success=Travis CI - Pull Request
        actions:
          queue:
            name: default

Multiple pull requests can be checked within one speculative check by settings
``batch_size``.

> For example, by settings ``speculative_checks: 2`` and ``batch_size: 3``,
> Mergify will create only 1 pull request to check if the first three enqueued
> pull requests are mergeable. And a second one to check the three next enqueued
> pull requests.

Reviewer: ```suggestion
For example, by settings ``speculative_checks: 2`` and ``batch_size: 3``,
Mergify will create two pull requests: a first one to check if the first three enqueued pull requests are mergeable, and a second one to check the three next enqueued pull requests.
```
